processing/src/main/java/org/apache/druid/query/rowsandcols/LazilyDecoratedRowsAndColumns.java

```diff
@@ -418,4 +418,116 @@ private Pair<byte[], RowSignature> naiveMaterialize(RowsAndColumns rac)
 
     return Pair.of(frameWriter.toByteArray(), sigBob.build());
   }
+
+
+  @Nullable
+  public Pair<byte[], RowSignature> naiveMaterializeToRowFrames(RowsAndColumns rac)
```

Comment:
Try to have a separate semantic interface like the WireTransferable. Let's call it FrameSerializable. Give it methods such a makeColumnFrame() and makeRowFrame(). Create a default implementation and then in operators.go ask using the as method to get the type of frame I want (row based or columnar).